Add a clarification note about linking the framework with the app (#11880)
DataDog/documentation: content/en/continuous_integration/setup_tests/swift.md

3. If you run UITests, also link the app running the tests with this library.

> <div class="alert alert-warning"><strong>Note</strong>:  The framework is only useful for testing and it should only be linked with the application when running tests. The framework should not be distributed to final users. </div>

Reviewer: ```suggestion
<div class="alert alert-warning"><strong>Note</strong>: This framework is useful only for testing and should only be linked with the application when running tests. Do not distribute the framework to your users. </div>
```
